lib/galaxy/jobs/runners/util/job_script/__init__.py

```diff
@@ -119,9 +119,8 @@ def _handle_script_integrity(path, config):
     sleep_amt = getattr(config, "check_job_script_integrity_sleep", DEFAULT_INTEGRITY_SLEEP)
     for i in range(count):
         try:
-            proc = subprocess.Popen([path], shell=True, env={"ABC_TEST_JOB_SCRIPT_INTEGRITY_XYZ": "1"})
-            proc.wait()
-            if proc.returncode == 42:
+            returncode = subprocess.call([path], env={"ABC_TEST_JOB_SCRIPT_INTEGRITY_XYZ": "1"})
```

Comment:
Any chance the exception happens before the inner try/except clause, and we just fail directly ?

Reply:
If the script doesn't exist we should get a `OSError: [Errno 2] No such file or directory`, which would be caught later on and look like the inetgrity verification failed.

Reply:
possibly? It would be nice to have a logging message on the exception.

Reply:
I'll open a PR.

Reply:
https://github.com/galaxyproject/galaxy/pull/4718 -- if it fails with anything except for `OSError` we should have a better idea, if it doesn't then this was probably the issue, and it would also explain why this test never worked for me on NFS

Reply:
https://github.com/galaxyproject/galaxy/pull/4719